FP-Erlang/Week 6/Assignment 5/Shashank/higherorder.erl

```diff
@@ -0,0 +1,118 @@
+%%%-------------------------------------------------------------------
+%%% @author shashankp
+%%% @copyright (C) 2017, <COMPANY>
+%%% @doc
+%%%
+%%% @end
+
+%%%-------------------------------------------------------------------
+-module(higherorder).
+-author("shashankp").
+
+%% API
+-export([add/1, subtract/1, multiply/1, divide/1, greaterThanX/1, lessThanX/1, fold/2, filter/1, map/1, all/1, takewhile/1, derive/2, integral/4, foldr/3, foldl/3]).
+
+% Question 1
+
+add(X) ->
+  fun(Y) -> X + Y end.
+
+subtract(X) ->
+  fun(Y) -> X - Y end.
+
+multiply(X) ->
+  fun(Y) -> X * Y end.
+
+divide(X) ->
+  fun(Y) -> X / Y end.
+
+greaterThanX(X) ->
+  fun(Y) -> Y > X end.
+
+lessThanX(X) ->
+  fun(Y) -> Y < X end.
+
```

Comment:
Add the two argument versions of the functions by making calls to the curried functiona